Initial Version

PythonicNinja opened the pull request.

dmydlo committed:
start

dmydlo committed:
project_start

dmydlo committed:
new start

dmydlo committed:
pet with basic click after chemotherapy

dmydlo committed:
add stop

dmydlo committed:
start_project python2&3 string formatting

dmydlo committed:
creating, no config.yaml for now, _lock

dmydlo committed:
repairs from pull-request reviews, creating basis of project_start/stop files

dmydlo committed:
export to bl.py, naive tests

dmydlo committed:
create with templates, tests with mock-ed Popen

dmydlo committed:
first fixture

dmydlo committed:
create.sh back to sh from bash, click nesting in edit

dmydlo committed:
small changes

dmydlo committed:
rewrite create to python

dmydlo committed:
project start without 'start', tasks: create, edit, run

dmydlo committed:
less shell and redundant Popens for editor access

dmydlo committed:
somehow it works

dmydlo committed:
./projects -> os.path.join(get_project_root.....

dmydlo committed:
hunt for old style string formating

dmydlo committed:
pet.py uses now get_projects_root()

dmydlo committed:
templates

dmydlo committed:
boot.sh (start) points to right shell_profiles

dmydlo committed:
instalation folder in config.py

dmydlo committed:
should create required pet/projects folder

dmydlo committed:
ok, without config this time

dmydlo committed:
basic register, archive and restore

dmydlo committed:
listing archive

dmydlo committed:
listing tasks, create -> project, task

dmydlo committed:
removing tasks

dmydlo committed:
tasks without turned on project - shitty way

dmydlo committed:
task run without project right way plus args

dmydlo committed:
test repairs

dmydlo committed:
list fix

dmydlo committed:
test_print_list

dmydlo committed:
Create as class plus less os.join, projects_root - more project_root)

dmydlo committed:
some tests

dmydlo committed:
51% -> 53%, testing Exceptions

dmydlo committed:
+4% (57%) test_task_exec_class

dmydlo committed:
(+21%) 57% -> 78% test_project_creator_class()

dmydlo committed:
(+8%) 78% -> 86%

dmydlo committed:
100% rakowe testy - polecam

dmydlo committed:
a lot of bad code in here, one page TODOs...

dmydlo committed:
futher changes

dmydlo committed:
deletion of redundant checking in cli

dmydlo committed:
bl.py order

dmydlo committed:
geting rid of some of redundant stuff

dmydlo committed:
pet.py clean complete

dmydlo committed:
small stuff

dmydlo committed:
completion should work

dmydlo committed:
complete yes, tests removed due to changes

dmydlo committed:
fixes

PythonicNinja committed:
ProjectLock is option for project start

dmydlo committed:
lockable checks if locked - lock blocks ONLY new instances of project

dmydlo committed:
templates are not projects

PythonicNinja committed:
some refactoring requests with TODO statement

dmydlo committed:
LESS CANCERSS

dmydlo committed:
start with generalShellMixin

dmydlo committed:
rewriten TODOs and other stuff, keeping things working

dmydlo committed:
working lockable :)

dmydlo committed:
Shell classes

dmydlo committed:
refactoring names

dmydlo committed:
generalization of tasks

dmydlo committed:
check start, start -l, task, task -i for _lockable and then merge

dmydlo committed:
working lockable decorator

dmydlo committed:
Merge of local branch

dmydlo committed:
full bash_complete

PythonicNinja commented on pet/complete.bash:
```diff
@@ -6,19 +6,65 @@ _pet()
     prev="${COMP_WORDS[COMP_CWORD-1]}"
     opts="archive clean edit init list register remove rename restore"
-    projects=""
+    projects=" new"
```
I think it shouldn't be there

PythonicNinja commented on pet/file_templates.py:
```diff
@@ -29,7 +29,7 @@ def cli(ctx, l):
 @click.option('-i', is_flag=True)
```
maybe also add `--interactive`
and change `i` to `interactive`

PythonicNinja commented on pet/complete.bash:
```diff
@@ -6,19 +6,65 @@ _pet()
-        opts="${opts} stop task"
-    else
         opts="${opts} run"
+        #             p->t
```
remove it

PythonicNinja commented on pet/bl.py:
```diff
@@ -465,7 +465,7 @@ def create_task(project_name, task_name):
+    task_file_path = os.path.join(project_root, "tasks", task_name + ".sh")
     Popen(["/bin/sh", "-c", "echo '#!/bin/sh' > {0}".format(task_file_path)]).communicate(input)
     edit_file(task_file_path)
     os.chmod(task_file_path, 0o755)
```
maybe intorduce `file_executable=0o755`

PythonicNinja commented on pet/cli.py:
```diff
@@ -222,7 +232,7 @@ def edit(name):
         """runs projects task"""
         try:
-            bl.run_task(project, task, active, i, args)
+            bl.run_task(project, task, "", i, args)
```
`project_name, task_name` and `""` is?

PythonicNinja replied:
same for `i`

PythonicNinja commented on pet/cli.py:
```diff
@@ -1,18 +1,22 @@
 
 cli = click.Group()
-active = os.environ.get('PET_ACTIVE_PROJECT', False)
+active = os.environ.get('PET_ACTIVE_PROJECT', '')
```
`default=""`

PythonicNinja commented on pet/make.py:
```diff
@@ -1,5 +1,10 @@
+# Python
```
remove comments

PythonicNinja replied:
rename file

PythonicNinja commented on pet/make.py:
```diff
@@ -11,10 +16,12 @@ def create_folders():
         os.makedirs(os.path.join(PET_FOLDER, "projects"))
     if not os.path.exists(os.path.join(PET_FOLDER, "archive")):
```
just call `makedirs`

dmydlo replied:
![obraz](https://cloud.githubusercontent.com/assets/22706074/24356431/3a168778-12fa-11e7-9b18-26d3fc8a10c8.png)

dmydlo committed:
run_task without active_project, create_task with any extention

dmydlo committed:
better handling of tasks

dmydlo committed:
applying comments

dmydlo committed:
before tests

dmydlo committed:
27th

PythonicNinja commented on pet/bl.py:
```diff
@@ -0,0 +1,569 @@
+def lockable(check_only=True, check_active=False, *args, **kwargs):
+    def _lockable(func, *args, **kwargs):
+        def __lockable(self=None, project_name='', check_only=check_only *args, **kwargs):
+            if check_only and kwargs.pop('lock'):
```
replace lines untill `kwargs.pop('lock')` with:
```
check_only = not kwargs.pop('lock', default=(not check_only)) 
```

PythonicNinja commented on pet/bl.py:
```diff
@@ -0,0 +1,569 @@
+            GeneralShellMixin.task_exec(self, project_name, task_name, interactive, args)
+
+
+@functools.lru_cache()
```
we need polyfill for that:
```
    @functools.lru_cache()
AttributeError: 'module' object has no attribute 'lru_cache'
```

PythonicNinja commented:
```
| => pet init -n pet1
sed: 1: "/Users/pythonicninja/Py ...": extra characters at the end of p command
```

PythonicNinja committed:
setup.py initial version,
support for osx brew completions,
refactored check_only for lockable decorator,
introduced main functions for startup scripts,

PythonicNinja commented:
```
|  ~/PycharmProjects/pet   (master) 
| => pet list -t
test
[pet1] (env) ________________________________________________________________________________
| pythonicninja 
|  ~/PycharmProjects/pet   (master) 
| => pet test
Traceback (most recent call last):
  File "/Users/pythonicninja/PycharmProjects/pet/env/bin/pet", line 11, in <module>
    load_entry_point('Pet', 'console_scripts', 'pet')()
  File "/Users/pythonicninja/PycharmProjects/pet/pet/cli.py", line 250, in main
    multi_cli()
  File "/Users/pythonicninja/PycharmProjects/pet/env/lib/python3.5/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/pythonicninja/PycharmProjects/pet/env/lib/python3.5/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/pythonicninja/PycharmProjects/pet/env/lib/python3.5/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/pythonicninja/PycharmProjects/pet/env/lib/python3.5/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/pythonicninja/PycharmProjects/pet/env/lib/python3.5/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/pythonicninja/.pet/projects/pet1/tasks.py", line 13, in test
    run_task("pet1", "test", interactive, args)
  File "/Users/pythonicninja/PycharmProjects/pet/pet/bl.py", line 551, in run_task
    get_shell().task_exec(project_name, task_name, interactive, args)
TypeError: __lockable() takes from 0 to 3 positional arguments but 5 were given
```

PythonicNinja commented:
```
| => pet --help
Usage: pet [OPTIONS] COMMAND [ARGS]...

Options:
  --help  Show this message and exit.

Commands:
  archive   archives project
  clean     unlocks all projects
  edit      edits task if given name else active project
  init      creates new project
  list      lists all projects
  register  registers .pet as project folder
  remove    removes task
  rename    renames task
  restore   restores project from archive
  stop      stops project
  task      creates new task
```

Is wrong because following line lists tasks:
```
| => pet list -t
```

PythonicNinja committed:
added support for `pet --version`

PythonicNinja committed:
FIX: multi_cli() got an unexpected keyword argument 'version'

dmydlo committed:
tasks are working as intended,
changed comment in cli.list,
new exception Info e.g. use alias will be avaible during next boot

dmydlo committed:
refactor of cli arguments

dmydlo committed:
correct checking name availability

dmydlo committed:
context manager for petexceptions

dmydlo committed:
backport of lru_cache

PythonicNinja committed:
added tox,
refactored tests folder,
isorted imports,

dmydlo committed:
19:21

PythonicNinja committed:
fix pet completions,
default is not present in python2

PythonicNinja committed:
version support for pet

PythonicNinja committed:
mock should use polyfills

dmydlo committed:
expand user

dmydlo committed:
lru_cache backport as requirement

dmydlo committed:
tasks with all commands

dmydlo committed:
small cleaning of old tests

dmydlo committed:
terminal tab name

PythonicNinja committed:
wait for process to stop

PythonicNinja committed:
Right now you can invoke it: pet {0}".format(task_name)

PythonicNinja committed:
TASK_NAME[.<extension>]

dmydlo committed:
Merge branch 'master' of github.com:dmydlo/pet

dmydlo committed:
tasks without extension

PythonicNinja committed:
PS1 export project name so that terminal tab can have info about project,
refactored templates

dmydlo committed:
executable files without extension where leaving *

dmydlo committed:
29th

dmydlo committed:
merge

dmydlo committed:
why not both

PythonicNinja committed:
code style tweaks

PythonicNinja commented on pet/exceptions.py:
```diff
@@ -1,8 +1,24 @@
 ]
 
 
+@unique
```
probably we would need compat https://pypi.python.org/pypi/enum34

dmydlo committed:
first docs

dmydlo committed:
q fix

dmydlo committed:
quick fix

dmydlo committed:
quick fix

dmydlo committed:
quick fix

dmydlo committed:
quick fix

dmydlo committed:
quick fix

dmydlo committed:
quick fix

dmydlo committed:
quick fix

dmydlo committed:
quick fix

dmydlo committed:
completion, installation

dmydlo committed:
config

dmydlo committed:
del todo

dmydlo committed:
50% tests

dmydlo committed:
61% tests

dmydlo committed:
64%

dmydlo committed:
69%

dmydlo committed:
80%

dmydlo committed:
quick fix

dmydlo committed:
numering terminal tabs

dmydlo committed:
in_place init, context_manager for active, stoped using pet_install_folder

dmydlo committed:
q fix of tasks

dmydlo committed:
q fix install scripts

dmydlo committed:
q fix install scripts

dmydlo committed:
q fix install scripts

dmydlo committed:
q fix install scripts

dmydlo committed:
setup changes

dmydlo committed:
setup changes

dmydlo committed:
quick change

dmydlo committed:
another try

dmydlo committed:
another try

dmydlo committed:
another try

dmydlo committed:
context manager test

dmydlo committed:
refactor of 2 tests

dmydlo committed:
keeping terminal tab name, edit-file without editor

dmydlo committed:
figthing sed error in task_remove

dmydlo committed:
pet - task_name auto_completion, redo of CLI -> no more project_name.py and tasks.py files

dmydlo committed:
kwierzbicki issue

dmydlo committed:
unintentional register

dmydlo committed:
NoneType issues

dmydlo committed:
install script

dmydlo committed:
Update README.md

dmydlo committed:
small tweak in install

dmydlo committed:
small tweak in install

dmydlo committed:
small tweak in install

dmydlo committed:
small tweak in install

dmydlo committed:
small tweak in install

dmydlo committed:
small tweak in install

dmydlo committed:
small tweak in install

dmydlo committed:
small tweak in install

dmydlo committed:
small tweak in install

dmydlo committed:
small tweak in install

dmydlo committed:
small tweak in install

dmydlo committed:
pip -upgrade stops process

dmydlo committed:
red warning etc

dmydlo committed:
green completion

dmydlo committed:
additional message what went wrong during installation

dmydlo committed:
pet init -i inconsistency

dmydlo committed:
big patch: pet clean is now pet remove -l, new way of finding version, raising errors when there are no standard (recreate) folders but were invoked, new projects counter, no more active_projects file, no more pet init -d - now it adds cd in start everytime

dmydlo committed:
removed creating active_projects file at installation

dmydlo committed:
pstree into ps aux

dmydlo committed:
install tweak

dmydlo committed:
edit config thru edit_file

dmydlo committed:
spliting shell_profiles into each for every shell

dmydlo committed:
first touches on zsh

dmydlo committed:
zsh completion based on bash completion

dmydlo committed:
some tweaks

dmydlo committed:
deploy in python

dmydlo committed:
quick fix

dmydlo committed:
quick fix

dmydlo committed:
quick fix

dmydlo committed:
quick fix

dmydlo committed:
quick fix

dmydlo committed:
quick fix

dmydlo committed:
tweaks in templates

dmydlo committed:
zsh tasks

dmydlo committed:
quick fix

dmydlo committed:
pet list --tree

dmydlo committed:
moved long string

dmydlo committed:
more file templates instead of other crap

dmydlo committed:
18th

dmydlo committed:
quick fix in deploy

dmydlo committed:
q fix

dmydlo committed:
tests good up to 150th line

dmydlo committed:
good tests up to 200th line

dmydlo committed:
Mocking methods of already mocked objects

PythonicNinja committed:
directory must end with '/'

dmydlo committed:
got rid of problems with / in paths by formatting paths in python not leaving {0}/sth in sh...

dmydlo committed:
Merge branch 'master' of github.com:dmydlo/pet

dmydlo committed:
we should leave it to os.path.join

dmydlo committed:
full test of lockable decorator

dmydlo committed:
GeneralShellMixin tests

dmydlo committed:
Bash class tests

dmydlo committed:
Zsh class tests

dmydlo committed:
another hopefully good tests

dmydlo committed:
another hopefully good tests

PythonicNinja commented on install.bash:
```diff
@@ -0,0 +1,49 @@
+if [ "$USER" == 'root' ]; then
+    printf "\n\n\e[1;31mWarning (used as root): During first run use\npet recreate\e[0m\n"
+else
+    create_folders
```
instead of `create_folders`, we can remove function of 
`create_folders` 
and 
`python 'pet-master/pet/cli.py' 'recreate'`

PythonicNinja commented on pet/file_templates.py:
```diff
@@ -0,0 +1,44 @@
+# add here shell code to be executed while entering project
+'''
+
+edit_file_popen_template = """
```
```
| => pet edit docs
/bin/sh: line 10: /usr/bin/nano: No such file or directory
```

PythonicNinja commented on setup.py:
```diff
@@ -0,0 +1,36 @@
+requirements = [
+    line.split('==')[0]
+    for line in open('requirements.txt', 'r').readlines()
+    ]
```
`]` should be at the very start of line

PythonicNinja commented on docs/source/user/basic/list.rst:
```diff
@@ -0,0 +1,13 @@
+
+- archived projects ``pet list -o``
+
+- all above ``pet list --tree``
```
Maybe we should use tabulate for that.
It's not looking perfect:
```
| => pet list --tree
breta
breta-admin
breta-webapp
pet1
 - docs - style - test
villoid-backend
villoid-infrastructure
villoid-merchant-api
villoid-merchant-web
villoid-store-api
villoid-store-web
 - lint
```

PythonicNinja commented on docs/source/user/basic/list.rst:
```diff
@@ -0,0 +1,13 @@
+Listing
+=======
+
+Pet provides listing of:
```
we lack command to list templates

PythonicNinja commented on docs/source/user/basic/list.rst:
```diff
@@ -0,0 +1,13 @@
+=======
+
+Pet provides listing of:
+
```
Also how to create template?

PythonicNinja replied:
I tried based on already existing project and it doesn't work also editor error is raised:
```
| => pet init -i -t pet1 -n pet-project
pet1 - is project not template
/bin/sh: line 10: /usr/bin/nano: No such file or directory
/bin/sh: line 10: /usr/bin/nano: No such file or directory
```

PythonicNinja commented:
I think that Pet should use `.pet` as it's init folder.

I find some problem due to that fact in terms of `start.sh` and going to specific route...
Solution to that would be to have relative paths in `start.sh` and export personal project location.

dmydlo committed:
install uses pet-s recreate

dmydlo committed:
bracket style in setup

dmydlo committed:
list templates

dmydlo committed:
checking existence of editors via posix command -v

dmydlo committed:
complete templates

dmydlo committed:
tests

dmydlo committed:
print tree

dmydlo committed:
local.entry and exit

dmydlo committed:
zsh auto-completion

dmydlo committed:
deploy

dmydlo committed:
fixes

dmydlo committed:
q fix

dmydlo committed:
changes in install

dmydlo committed:
install

dmydlo committed:
tmp 4 locals

dmydlo committed:
install directory

dmydlo committed:
locals for start/stop

dmydlo committed:
local tasks

dmydlo committed:
locals for tasks

dmydlo committed:
print active projects

dmydlo committed:
print tasks without local entry and exit

dmydlo committed:
prompt with task name and tasks with all ext once again

dmydlo committed:
prompt and complete

dmydlo committed:
tests repaired

dmydlo committed:
docs

dmydlo committed:
start.sh fix

dmydlo committed:
docs

dmydlo committed:
pet project

dmydlo committed:
no bashrc

dmydlo committed:
task exec fix

dmydlo committed:
gif

dmydlo committed:
isort
